dsc/exceptions.py

```diff
@@ -12,3 +12,7 @@ class InvalidWorkflowNameError(Exception):
 
 class ItemMetadatMissingRequiredFieldError(Exception):
     pass
+
+
+class ReconcileError(Exception):
+    pass
```

Comment:
This is to differentiate the following scenarios: 

1. If `reconcile` is called but it is not applicable for a workflow, raise `TypeError`.
2. If `reconcile` is called and it is applicable for a workflow _and_ there are items missing either bitstreams or metadata, raise `Reconcile` error.

Reply:
I like this approach and it sets the app up for more thorough automation in the future as we've discussed!